actors/evm/src/interpreter/stack.rs

```diff
@@ -1,62 +1,93 @@
-#![allow(dead_code)]
+#![allow(dead_code, clippy::missing_safety_doc)]
 
-use {crate::interpreter::U256, arrayvec::ArrayVec, serde::Serialize};
+use crate::interpreter::U256;
 
 /// Ethereum Yellow Paper (9.1)
-pub const MAX_STACK_SIZE: usize = 1024;
+pub const STACK_SIZE: usize = 1024;
+
+const INITIAL_STACK_SIZE: usize = 32;
 
 /// EVM stack.
-#[derive(Clone, Debug, Default, Serialize)]
-pub struct Stack(pub ArrayVec<U256, MAX_STACK_SIZE>);
+#[derive(Clone, Debug)]
+pub struct Stack {
+    sk: Vec<U256>,
+    d: usize,
+}
 
 impl Stack {
     #[inline]
-    pub const fn new() -> Self {
-        Self(ArrayVec::new_const())
+    pub fn new() -> Self {
+        Stack { sk: Vec::from([U256::zero(); INITIAL_STACK_SIZE]), d: 0 }
     }
 
     #[inline]
-    const fn get_pos(&self, pos: usize) -> usize {
-        self.len() - 1 - pos
+    pub fn require(&self, required: usize) -> bool {
+        required <= self.d
```

Comment:
Ok, I should have reviewed this more carefully.

What is d? Let's use actual names please?

Reply:
d is for depth.